### PR TITLE
feat(rust, python): lazy diagonal concat.

### PR DESCRIPTION
**Pull request by AnatolyBuga**
diag_concat exists for Eager frames, but not for Lazy frames. This function is a diag_concat for lazy frames.

PS: I've been trying to test this with `cargo test test_diag_concat_lf --features=diagonal_concat,lazy` but couldn't compile due to errors. Ran this exact test locally and it passed.

### CODE REVIEW

**Comment by ritchie46** [comment]
Thanks @AnatolyBuga. Nice solution really great how you could make it work with existing functionality. :+1:

**Comment by ritchie46** [comment]
@AnatolyBuga we still have to investigate you solution as it does not work the same as the `diagonal` version on `DataFrame`s.

This test fails at the moment:

```python
    a = pl.DataFrame({"a": [1, 2]})
    b = pl.DataFrame({"b": ["a", "b"], "c": [1, 2]})
    c = pl.DataFrame({"a": [5, 7], "c": [1, 2], "d": [1, 2]})

    for out in [
        pl.concat([a, b, c], how="diagonal"),
        pl.concat([a.lazy(), b.lazy(), c.lazy()], how="diagonal").collect(),
    ]:
        expected = pl.DataFrame(
            {
                "a": [1, 2, None, None, 5, 7],
                "b": [None, None, "a", "b", None, None],
                "c": [None, None, 1, 2, 1, 2],
                "d": [None, None, None, None, 1, 2],
            }
        )

        assert out.frame_equal(expected, null_equal=True)
```

File: py-polars/src/lib.rs [hunk 1]
```diff
@@ -366,6 +366,23 @@ fn py_diag_concat_df(dfs: &PyAny) -> PyResult<PyDataFrame> {
...
+
+    let lfs = iter
+        .map(|item| {
+            let item = item?;
+            get_lf(item)
+        })
+        .collect::<PyResult<Vec<_>>>()?;
+
+    let lf =
+        polars::lazy::dsl::functions::concat(lfs, rechunk, parallel).map_err(PyPolarsErr::from)?;
```
Comment (ghuls) [inline py-polars/src/lib.rs:382]: Should it not be? `diag_concat_lf`

Comment (ritchie46) [reply py-polars/src/lib.rs:382]: Thanks.. My bad. 🙈

**Comment by StijnKas** [comment]
Hi, very useful as I've ran into mismatches in column names when reading multiple files quite frequently! 

Will there also be a way to call this when doing a `scan` over multiple files and then collecting them? Naturally a workaround would be to create all scan queries using a for loop over the list of files and then `concat` and `collect` just like in the example test, but it would be convenient if there was something like this:

`pl.scan_ipc(path/files*.ipc).collect(how='diagonal')`

**Pull request merged**